server/lib/transactions.ts

```diff
@@ -57,140 +56,142 @@ export function getTransactions(collectiveids, startDate = new Date('2015-01-01'
-    hostCurrencyFxRate = 1 / parseFloat(currencyConversion.exchangeRate); // paypal returns a float from host.currency to expense.currency
-    paymentProcessorFeeInHostCurrency = Math.round(hostCurrencyFxRate * paymentProcessorFeeInCollectiveCurrency);
-  }
-  // PayPal Payouts
-  else if (payoutMethodType === PayoutMethodTypes.PAYPAL && transactionData?.payout_batch_id) {
-    hostCurrencyFxRate = transactionData.currency_conversion?.exchange_rate
-      ? 1 / toNumber(transactionData.currency_conversion?.exchange_rate)
-      : await getFxRate(expense.currency, host.currency, expense.incurredAt || expense.createdAt);
+/**
+ * A function to create transactions for a given expense that is agnostic of the payout method
+ *
+ * TODO: This function should accept an `amount` to automatically represent what was really paid from the payment provider,
+ * in case it differs from expense.amount (e.g. when fees are put on the payee)
+ */
+export async function createTransactionsFromPaidExpense(
+  host,
+  expense,
+  fees: FEES_IN_HOST_CURRENCY = DEFAULT_FEES,
+  /** Set this to a different value if the expense was paid in a currency that differs form the host's */
+  expenseToHostFxRateConfig: number | 'auto',
+  /** Will be store in transaction.data */
+  transactionData: Record<string, unknown> = null,
+  /** @deprecated Only used for paypal adaptive, to link the payment method */
+  paymentMethod = null,
+) {
+  fees = { ...DEFAULT_FEES, ...fees };
+  expense.collective = expense.collective || (await models.Collective.findByPk(expense.CollectiveId));
 
-    paymentProcessorFeeInCollectiveCurrency = round(toNumber(transactionData.payout_item_fee?.value) * 100);
-    paymentProcessorFeeInHostCurrency = Math.round(hostCurrencyFxRate * paymentProcessorFeeInCollectiveCurrency);
-    hostFeeInCollectiveCurrency = Math.round((1 / hostCurrencyFxRate) * hostFeeInHostCurrency);
-    platformFeeInCollectiveCurrency = Math.round((1 / hostCurrencyFxRate) * platformFeeInHostCurrency);
-  } else if (payoutMethodType === PayoutMethodTypes.BANK_ACCOUNT) {
-    if (host.settings?.transferwise?.ignorePaymentProcessorFees) {
-      paymentProcessorFeeInHostCurrency = 0;
-    } else if (transactionData?.paymentOption?.fee?.total) {
-      paymentProcessorFeeInHostCurrency = Math.round(transactionData.paymentOption.fee.total * 100);
-    }
-    // Notice this is the FX rate between Host and Collective, the user is not involved here and that's why TransferWise quote rate is irrelevant here.
-    hostCurrencyFxRate = await getFxRate(expense.currency, host.currency);
-    paymentProcessorFeeInCollectiveCurrency = Math.round((1 / hostCurrencyFxRate) * paymentProcessorFeeInHostCurrency);
-    hostFeeInCollectiveCurrency = Math.round((1 / hostCurrencyFxRate) * hostFeeInHostCurrency);
-    platformFeeInCollectiveCurrency = Math.round((1 / hostCurrencyFxRate) * platformFeeInHostCurrency);
+  // Get the right FX rate for the expense
+  let expenseToHostFxRate;
```

Comment:
🍨 I rather use a ternary and declare `expenseToHostFxRate` as a constant.

```suggestion
  const expenseToHostFxRate = expenseToHostFxRateConfig === 'auto'
    ? await getFxRate(expense.currency, host.currency, expense.incurredAt || expense.createdAt)
    : expenseToHostFxRateConfig;
```